src/bigint.cpp

```diff
@@ -230,7 +235,201 @@ Bigint &Bigint::operator*=(int const &b)
+      int x = number[0];
+      if(x%2 == 0)
+        return true;
+      else
+          return false;
+    }
+
+    else if(number.size()>1){
+      int x = *(number.begin());
+      if(x%2 == 0)
+        return true;
+      else
+          return false;
+    }
+
+}
+
+
+Bigint Bigint::sub_number(Bigint &p, Bigint &q){
+
+      std::string tmpx0, tmpx1, tmpx3;
+      long window_size = q.digits();
+      int last_i_iterator;
+      std::vector<int>::iterator i=p.number.end()-1;
+
+      if(window_size>p.digits())
+          window_size=p.digits();
+
+
+      while(i>=p.number.begin() && window_size>0){
+
+          tmpx0 = to_string(*i);
+          int ssss = tmpx0.size();
+          if(i!=p.number.end()-1 && ssss<9){
+              tmpx0 = std::string(9-ssss,'0') + tmpx0;
+          }
+
+          if(window_size - ssss < 0){
+              tmpx3=tmpx0;
+              tmpx0="";
+              for(int i=0;i<window_size;i++){
+                  tmpx0=tmpx0+tmpx3[i];
+                  last_i_iterator=i;
+              }
+              window_size = window_size - tmpx0.size();
+              tmpx1 = tmpx1 + tmpx0;
+          }
+          else{
+              window_size = window_size - tmpx0.size();
+              tmpx1 = tmpx1 + tmpx0;
+              i--;
+          }
+      }
+
+      Bigint c(tmpx1);
+
+      if(c<q && i>=p.number.begin()){
+          if(tmpx3.size()!=0){
+              window_size++;
+              tmpx0 = tmpx3[last_i_iterator+1];
+              tmpx1 = tmpx1 + tmpx0;
+              c=tmpx1;
+          }
+          else{
+              window_size++;
+              tmpx0 = to_string(*i);
+              if(i!=p.number.end()-1 && tmpx0.size()<9){
+                  tmpx0 = std::string(9-tmpx0.size(),'0') + tmpx0;
+              }
+              tmpx0 = tmpx0[0];
+              tmpx1 = tmpx1 + tmpx0;
+              c=tmpx1;
+          }
+
+      }
+
+      return c;
+}
+//Division
+
+std::vector<Bigint> Bigint::divide(Bigint q){
+
+    /*Algorithm used is "Double division algorithm"*/
+
+    Bigint p = *this;
+    std::vector<Bigint> answer;
+    int look_up_table_size=4;
+    std::vector<Bigint> look_up(look_up_table_size);
+    bool done_flag=false ;
+    Bigint tmp_quotient, sum_quotient, tmpx1, tmpx2;
+
+    Bigint zero("0");
+
+    if(q==zero || q == 0){
+    //std::cout << "Error: Dividing by zero" << std::endl;
+    //exit (EXIT_FAILURE);
+    throw "Dividing by zero!";
+    }
+
+    bool this_sign = this -> positive;
+    bool q_sign = q.positive;
+
+    p.positive = true;
+    q.positive = true;
+
+
+    look_up[0]=q;
+    look_up[1]=q*2;
+    look_up[2]=q*4;
+    look_up[3]=q*8;
+
+
+    while(true){
+
+       Bigint sub_p=sub_number(p, q);  // cut a portion from the dividened equal in size with the divisor
+
+        for(int i=0;i<look_up_table_size;i++){ // looking in the look_up table
+
+            if(sub_p<look_up[i] && i!=0){
+                tmpx1=look_up[i-1];
+                tmp_quotient = std::pow(2,i-1);
```

Comment:
Maybe <code>1<<(i-1)</code> is an easier and faster choice. <code>pow</code> in <code>cmath</code> is for float number